File: cache/src/main/java/com/osrsd/cache/loader/SpotAnimationLoader.java
package com.osrsd.cache.loader;

import com.displee.cache.index.Index;
import com.displee.cache.index.archive.Archive;
import com.osrsd.cache.Configs;
import com.osrsd.cache.Indexes;
import com.osrsd.cache.Library;
import com.osrsd.cache.def.Definition;
import com.osrsd.cache.def.SpotAnimationDefinition;
import com.osrsd.cache.provider.SpotAnimationProvider;
import com.osrsd.cache.util.Serializable;

import java.nio.ByteBuffer;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Objects;

public class SpotAnimationLoader implements Loader {

    @Override
    public Serializable load(Library library) {
        Index index = library.index(Indexes.CONFIG);
        Archive archive = index.archive(Configs.SPOT_ANIMS);

        assert archive != null;
        List<Definition> definitions = new ArrayList<>(archive.fileIds().length);
        Arrays.stream(archive.fileIds()).forEach(fileId -> {
            byte[] data = Objects.requireNonNull(archive.file(fileId)).getData();
            if (data != null) {
                definitions.add(SpotAnimationProvider.decode(ByteBuffer.wrap(data), new SpotAnimationDefinition(fileId)));
            }
        });
        return new Serializable(this, definitions, "/spotanims");
    }

}

File: app/src/main/kotlin/com/osedit/cache/loader/SpotAnimationLoader.kt
package com.osedit.cache.loader

import com.displee.cache.index.archive.Archive
import com.osedit.cache.Configs
import com.osedit.cache.Indexes
import com.osedit.cache.Library
import com.osedit.cache.provider.SpotAnimationProvider
import com.osedit.cache.util.Serializable
import com.osedit.spring.domain.SpotAnimationDefinition
import java.nio.ByteBuffer

class SpotAnimationLoader : Loader {

    override fun load(): Serializable {
        val archive: Archive = Library.index(Indexes.CONFIG).archive(Configs.SPOT_ANIMS)!!
        val definitions = archive.fileIds().map {
            SpotAnimationProvider.decode(ByteBuffer.wrap(archive.file(it)?.data), SpotAnimationDefinition(it))
        }
        return Serializable(this, definitions, "/spotanimations")
    }

}